src/components/backend-ai-session-view.ts

```diff
@@ -162,9 +215,204 @@ export default class BackendAiSessionView extends BackendAIPage {
+    menu.anchor = button;
+    if (!menu.open) {
+      menu.show();
+    }
+  }
+
+  _openExportToCsvDialog() {
     let menu = this.shadowRoot.querySelector("#dropdown-menu");
-    menu.open = !menu.open;
+    if (menu.open) {
+      menu.close();
+    }
+    console.log("Downloading CSV File...");
+    this._defaultFileName = this._getDefaultCSVFileName();
+    this.exportToCsvDialog.show();
+  }
+
+  _getFirstDateOfMonth() {
+    let date = new Date();
+    return new Date(date.getFullYear(), date.getMonth(), 2).toISOString().substring(0, 10);
+  }
+
+  _getDefaultCSVFileName() {
+    let date = new Date().toISOString().substring(0, 10);
+    let time = new Date().toTimeString().slice(0, 8).replace(/:/gi, '-');
+    return date + '_' + time;
+  }
+
+  /**
+   * Check date-to < date-from.
+   * */
+  _validateDateRange() {
+    let dateTo = this.shadowRoot.querySelector('#date-to');
+    let dateFrom = this.shadowRoot.querySelector('#date-from');
+
+    if (dateTo.value && dateFrom.value) {
+      let to = new Date(dateTo.value).getTime();
+      let from = new Date(dateFrom.value).getTime();
+      if (to < from) {
+        dateTo.value = dateFrom.value;
+      }
+    }
+  }
+
+  /**
+   * Scale the time in units of D, H, M, S, and MS.
+   *
+   * @param {number} value - time to want to scale
+   * */
+  _automaticScaledTime(value: number) { // number: msec.
+    let result = Object();
+    let unitText = ['D', 'H', 'M', 'S'];
+    let unitLength = [(1000 * 60 * 60 * 24), (1000 * 60 * 60), (1000 * 60), 1000];
+
+    for (let i = 0; i < unitLength.length; i++) {
+      if (Math.floor(value / unitLength[i]) > 0) {
+        result[unitText[i]] = Math.floor(value / unitLength[i]);
+        value = value % unitLength[i];
+      }
+    }
+    if (Object.keys(result).length === 0) { // only prints msec. when time is shorter than 1sec.
+      if (value > 0) {
+        result = {'MS': value};
+      } else { // No data.
+        result = {'NODATA': 1};
+      }
+    }
+    return result;
+  }
+
+  _msecToSec(value) {
+    return Number(value / 1000).toFixed(0);
+  }
+
+  _bytesToMB(value) {
+    return Number(value / (1024 * 1024)).toFixed(1);
+  }
+
+  _exportToCSV() {
+    const fileNameEl = this.shadowRoot.querySelector('#export-file-name');
+
+    if (!fileNameEl.validity.valid) {
+      return;
+    }
+    const exportList: any = [];
+
+    // Parameters
+    let status: any = ["RUNNING", "RESTARTING", "TERMINATING",  "PENDING", "PREPARING", "PULLING", "TERMINATED", "CANCELLED", "ERROR"];
+    if (globalThis.backendaiclient.supports('detailed-session-states')) {
+      status = status.join(',');
+    }
+    const fields = ["id", "name", "image", "created_at", "terminated_at", "status", "status_info", "access_key"];
+    if (this._connectionMode === "SESSION") {
+      fields.push("user_email");
+    }
+    if (globalThis.backendaiclient.is_superadmin) {
+      fields.push("containers {container_id agent occupied_slots live_stat last_stat}");
```

Comment:
@lizable  Does 20.03 support containers {} ? Please check it and add .support conditional flag for 20.03 compatibility. If it supports, it is ok.

Reply:
I checked and found graphql error related to field value, and fixed it.
After applying few commits, I confirmed that It works well in 20.03 with containers.